enterprise/web.jspparser/src/org/netbeans/modules/web/jsps/parserapi/TagAttributeInfo.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.web.jsps.parserapi;
+
+public class TagAttributeInfo {
+    private String name;
+    private String typeName;
+    private boolean required;
+    private boolean fragment;
+    private boolean canBeRequestTime;
+
+    public TagAttributeInfo() {
+    }
```

Comment:
question: aren't those `Tag*` objects in this package immutable once created?

Since its API, it might be better to remove the no-arg constructor and the setters - or even make it a record?